core/src/main/scala/kafka/message/MessageSet.scala

```diff
@@ -54,6 +54,22 @@ object MessageSet {
    */
   def entrySize(message: Message): Int = LogOverhead + message.size
 
+  /**
+   * Validate the "magic" values of messages are the same in a compressed message set and return the max timestamp
+   * of the inner messages.
+   */
+  def validateMagicValuesAndGetTimestamp(messages: Seq[Message]): Long = {
```

Comment:
Would it be better to return a MagicAndTimestamp so that the caller doesn't have to iterate the message set again to get the magic?